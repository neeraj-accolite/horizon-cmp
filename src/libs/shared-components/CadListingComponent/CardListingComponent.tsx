import React from 'react';
import { CardListingComponentProps } from './CardListingComponent.model';
import { Card } from '../Card';

const CardListingComponent: React.FC<CardListingComponentProps> = ({
  title,
  cards,
  backgroundColor,
}) => {
  return (
    <div className="flex flex-col">
      <h2 className="pb-16 pt-8 text-center text-2xl font-bold">{title}</h2>
      <div className="flex flex-wrap justify-start">
        {cards.map((card, index) => (
          <div key={index} className="box-border w-full p-2 sm:w-1/2 lg:w-1/3">
            <Card style={card.style}>
              <h3
                className="flex justify-center rounded-t-lg px-6 pb-2 pt-6 text-xl font-bold"
                style={{ backgroundColor: backgroundColor || '#F5F6F6' }}
              >
                {card.title}
              </h3>
              <div className="px-6 pb-12 pt-4">{card.children}</div>
            </Card>
          </div>
        ))}
      </div>
    </div>
  );
};

export default CardListingComponent;
