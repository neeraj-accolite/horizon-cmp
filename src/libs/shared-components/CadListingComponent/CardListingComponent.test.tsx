import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import CardListingComponent from './CardListingComponent';
import { CardListingComponentProps } from './CardListingComponent.model';

describe('CardListingComponent', () => {
  const defaultProps: CardListingComponentProps = {
    title: 'Our Plumbing Services',
    cards: [
      {
        title: 'Water Heaters',
        children: (
          <div>
            Keep your hot water flowing with our expert repair, proactive
            maintenance, and full installation services.
          </div>
        ),
        style: {
          borderColor: '#D9D9D9',
          borderWidth: '1px',
          borderRadius: '8px',
          borderStyle: 'solid',
        },
      },
      {
        title: 'Kitchen & Bathroom',
        children: (
          <div>
            Our expertise in repairing and installing kitchen and bathroom
            fixtures ensures these workhorses will have your back for years to
            come.
          </div>
        ),
        style: {
          borderColor: '#D9D9D9',
          borderWidth: '1px',
          borderRadius: '8px',
          borderStyle: 'solid',
        },
      },
    ],
    backgroundColor: '#F5F6F6',
  };

  test('renders the component with default props', () => {
    const { getByText } = render(<CardListingComponent {...defaultProps} />);
    expect(getByText('Our Plumbing Services')).toBeInTheDocument();
    expect(getByText('Water Heaters')).toBeInTheDocument();
    expect(getByText('Kitchen & Bathroom')).toBeInTheDocument();
  });

  test('renders card titles with the correct background color', () => {
    const { getByText } = render(<CardListingComponent {...defaultProps} />);
    const waterHeatersTitle = getByText('Water Heaters');
    const kitchenBathroomTitle = getByText('Kitchen & Bathroom');
    expect(waterHeatersTitle).toHaveStyle('backgroundColor: #F5F6F6');
    expect(kitchenBathroomTitle).toHaveStyle('backgroundColor: #F5F6F6');
  });

  test('renders card children correctly', () => {
    const { getByText } = render(<CardListingComponent {...defaultProps} />);
    expect(
      getByText(
        'Keep your hot water flowing with our expert repair, proactive maintenance, and full installation services.',
      ),
    ).toBeInTheDocument();
    expect(
      getByText(
        'Our expertise in repairing and installing kitchen and bathroom fixtures ensures these workhorses will have your back for years to come.',
      ),
    ).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<CardListingComponent {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
