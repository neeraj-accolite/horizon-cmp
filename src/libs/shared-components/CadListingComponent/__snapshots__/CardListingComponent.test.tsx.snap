// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardListingComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col"
  >
    <h2
      class="pb-16 pt-8 text-center text-2xl font-bold"
    >
      Our Plumbing Services
    </h2>
    <div
      class="flex flex-wrap justify-start"
    >
      <div
        class="box-border w-full p-2 sm:w-1/2 lg:w-1/3"
      >
        <div
          style="border-color: #d9d9d9; border-width: 1px; border-radius: 8px; border-style: solid;"
        >
          <h3
            class="flex justify-center rounded-t-lg px-6 pb-2 pt-6 text-xl font-bold"
            style="background-color: rgb(245, 246, 246);"
          >
            Water Heaters
          </h3>
          <div
            class="px-6 pb-12 pt-4"
          >
            <div>
              Keep your hot water flowing with our expert repair, proactive maintenance, and full installation services.
            </div>
          </div>
        </div>
      </div>
      <div
        class="box-border w-full p-2 sm:w-1/2 lg:w-1/3"
      >
        <div
          style="border-color: #d9d9d9; border-width: 1px; border-radius: 8px; border-style: solid;"
        >
          <h3
            class="flex justify-center rounded-t-lg px-6 pb-2 pt-6 text-xl font-bold"
            style="background-color: rgb(245, 246, 246);"
          >
            Kitchen & Bathroom
          </h3>
          <div
            class="px-6 pb-12 pt-4"
          >
            <div>
              Our expertise in repairing and installing kitchen and bathroom fixtures ensures these workhorses will have your back for years to come.
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
